### PR TITLE
Add line segment-line segment intersection

### PR DESCRIPTION
**Pull request by shehzan10**
Find the intersection between 2 line segments in 2D.

### CODE REVIEW

File: Source/Core/Intersections2D.js [hunk 1]
```diff
@@ -278,5 +280,76 @@ define([
...
+     * @returns {Cartesian2} The intersection point, undefined if there is no intersection point or lines are coincident.
+     *
+     * @example
+     * var result = Cesium.Intersections2D.computeLineSegmentLineSegmentIntersection(0.0, 0.0, 0.0, 2.0, -1, 1, 1, 1);
+     * // result === new Cesium.Cartesian2(0.0, 1.0);
+     */
+    Intersections2D.computeLineSegmentLineSegmentIntersection = function(x00, y00, x01, y01, x10, y10, x11, y11, result) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(x00)) {
+            throw new DeveloperError('x00 is required.');
```
Comment (hpinkos) [inline Source/Core/Intersections2D.js]: Use `Check.typeOf.number('x00', x00)` instead

**Comment by hpinkos** [comment]
@shehzan10 merge in master.  Other than that one comment, this looks good!

**Comment by shehzan10** [comment]
@hpinkos Updated.

**Comment by hpinkos** [comment]
Thanks @shehzan10 !

**Pull request merged**